[Gerald Neo] iP

geraldneo567 opened the pull request:
### DukePro
>"Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you nee to do. It's,
- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2122s2.github.io/website/schedule/week4/project.html) 
2. double-click it
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
  public static void main(String[] args) {
     Application.launch(MainApp.class, args);
  }
}
```

geraldneo567 committed:
Added Level-1 Functionality to Duke

geraldneo567 committed:
Added Level-2 Functionality

geraldneo567 committed:
Added Level-3 Functionality

geraldneo567 committed:
Added Level-4 Functionality and fixed Level-3 functionality

geraldneo567 committed:
Added Level-5 Functionality

geraldneo567 committed:
Added Level-6 Functionality

geraldneo567 committed:
Added text ui test

geraldneo567 committed:
added test.txt

geraldneo567 committed:
Deleted outdated runtest.bat

geraldneo567 committed:
Removed unnecessary test.txt file

geraldneo567 committed:
Added Level 7 functionality

geraldneo567 committed:
Added Level-8 Functionality

geraldneo567 committed:
Merge branch 'branch-Level-8'

geraldneo567 committed:
Made code more OOP

geraldneo567 committed:
Added duke package

geraldneo567 committed:
Added basic Junit tests for TaskList

geraldneo567 committed:
Added Junit tests for Parser

geraldneo567 committed:
Added JavaDoc Comments

geraldneo567 committed:
Tweak to comply with coding standard

geraldneo567 committed:
Add Level-9 Functionality

geraldneo567 committed:
Merge branch 'branch-A-CodingStandard'

geraldneo567 committed:
Merge branch 'branch-Level-9'

geraldneo567 committed:
Add gradle support

geraldneo567 committed:
Add basic GUI

geraldneo567 committed:
Add GUI imports

B1LLP4RK commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,26 @@
+ * Represents an instance of the <code>Duke</code> chatbot, which
+ * can then be run.
+ */
+
```
You can delete the trailing blank lines before the class definition:)

B1LLP4RK commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,26 @@
+        new Duke("../../../data/duke.txt");
+        Application.launch(Ui.class, args);
+        Duke.storage.save(Duke.tasks);
+
```
You can delete this blank line too:)

B1LLP4RK commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,21 @@
+ * a Task which contains a <code>String</code> time field corresponding
+ * to when the event occurs.
+ */
+
+
```
You can delete blank lines where appropriate, in all files:)

B1LLP4RK commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,42 @@
+
+    /**
+     * Parses input string to return appropriate enum type.
+     * @param input
```
I think the description can be a bit more concrete. i.e. the string input that the user types

B1LLP4RK commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,83 @@
+            for (Task t : ls) {
+                fw.write(t.saveString() + "\n");
+            }
+            fw.close();
```
I suggest closing `File` and `FileWriter` in finally block. This [link](https://stackoverflow.com/questions/37037718/is-it-necessary-to-call-close-in-a-finally-when-writing-files-in-java) should help:D I noticed similar issue in other places as well.

B1LLP4RK commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,83 @@
+
+    /**
+     * Loads contents of current saved file into tasklist.
+     * @return The list containing the pre-saved tasks.
```
I suggest having a line between the description and the parmeter section, as written [here](https://se-education.org/guides/conventions/java/intermediate.html#:~:text=Empty%20line%20between%20description%20and%20parameter%20section):D I noticed the same issue in other places too.

B1LLP4RK commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,119 @@
+     * Lists the string representation of all tasks currently in
+     * the list.
+     */
+    public void listTasks() {
```
I like how you always use well-read and intuitive names for the variables and methods 👍

B1LLP4RK commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,119 @@
+
+        for (int i = 0; i < ls.size(); i++) {
+            if (ls.get(i).getName().contains(keyword)) {
+                result +=(i + 1) + ". " + ls.get(i) + "\n";
```
This sounds a bit picky, but I suggest adding a space after the equals sign. For such things, I suggest finding the IDE shortcut to autoformat all the files in your project:)

B1LLP4RK commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,119 @@
+
+
+public class TaskList {
+    private List<Task> ls;
```
ls is good, but I think list is slightly better in terms of readability 🤔

B1LLP4RK commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,119 @@
+        }
+    }
+
+    public String findAndGetTasks(String keyword) {
```
I have a feeling that the method name can be more succinct. Perhaps `filter`?

B1LLP4RK reviewed (commented):
Concise and easy to read code overall🥇 Just some nits to fix :D

zechajw commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,42 @@
+     * @param input
+     * @return The appropriate enum type.
+     */
+    public CommandType parse(String input) {
```
Maybe you could use a switch-case block here rather than an if-else block to increase readability?

zechajw commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,71 @@
+
+
+public class Task {
+    private boolean completed;
```
Perhaps this variable name should be isCompleted rather than completed to better adhere to the coding convention?

zechajw commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,119 @@
+
+    /**
+     * Prints tasks in the list that has name containing the specified keyword.
+     * @param keyword
```
Perhaps you could add a description for each javadoc variable to enhance readability?

zechajw commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,119 @@
+    }
+
+    /**
+     * Utility getter method to obtain the underlying list.
```
I think that there is another way to write the javadoc comments. 

> Write the first sentence as a short summary of the method, as Javadoc automatically places it in the method summary table (and index).
In method header comments, the first sentence should start in the form Returns ..., Sends ..., Adds ... (not Return or Returnning etc.)

zechajw reviewed (commented):
Good work with the code so far!

I made some minor suggestions for code style but otherwise I think the code is good overall!

geraldneo567 committed:
Duke class and Parser class: added assertions
Duke class and Parser class: added assertions
For the Duke class, this is to ensure we successfully perform the load
of existing data into the program, which can be checked using
the assert.

For the Parser class, this is to prevent NullPointer exceptions.

geraldneo567 committed:
TaskList and Storage classes: Improved code quality

Improved the code quality for these classes by reducing
the complexity of the expressions.

geraldneo567 committed:
Merge pull request #2 from geraldneo567/branch-A-CodeQuality

Improve code quality

geraldneo567 committed:
Add update functionality

geraldneo567 committed:
Merge pull request #3 from geraldneo567/branch-C-Update

Add update functionality

geraldneo567 committed:
Update README.md

geraldneo567 committed:
Update README.md

geraldneo567 committed:
Add Ui.png

geraldneo567 committed:
Merge branch 'master' of https://github.com/geraldneo567/ip

geraldneo567 committed:
Update iP after peer evaluation

geraldneo567 committed:
Update README.md

geraldneo567 committed:
Update README.md

geraldneo567 committed:
Update README.md

geraldneo567 committed:
Reduce length of parse method

geraldneo567 committed:
Merge branch 'master' of github.com:geraldneo567/ip
